packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -701,15 +701,8 @@ function updateStaticChildren(c1: VNodes, c2: VNodes, parent: ParentNode) {
         if (n2 !== n1) {
             if (isVNode(n1)) {
                 if (isVNode(n2)) {
-                    if (isSameVnode(n1, n2)) {
-                        // both vnodes are equivalent, and we just need to patch them
-                        patch(n1, n2);
-                    } else {
-                        // TODO [#2697]: it is possible for the template to define key attributes outside of an
-                        //  iteration, leading to a situation where static children have different keys.
-                        unmount(n1, parent, true);
-                        mount(n2, parent, anchor);
-                    }
```

Comment:
This code is now unnecessary! And we have a test to prove it.

Previous discussion: https://github.com/salesforce/lwc/pull/2677#discussion_r808534831